benches/benches/bevy_ecs/index/index_iter_naive.rs

```diff
@@ -0,0 +1,35 @@
+use bevy_ecs::{prelude::*, system::SystemId};
+use core::hint::black_box;
+use glam::*;
+
+const PLANETS: u16 = 1_000;
+const SPAWNS: usize = 1_000_000;
+
+#[derive(Component, Copy, Clone, PartialEq, Eq, Hash)]
+#[component(immutable)]
+struct Planet(u16);
+
+fn find_planet_zeroes_naive(query: Query<&Planet>) {
+    for planet in query.iter().filter(|&&planet| planet == Planet(0)) {
+        let _ = black_box(planet);
+    }
+}
+
+pub struct Benchmark(World, SystemId);
+
+impl Benchmark {
+    pub fn new() -> Self {
+        let mut world = World::new();
+
+        world.spawn_batch((0..PLANETS).map(Planet).cycle().take(SPAWNS));
+
+        let id = world.register_system(find_planet_zeroes_naive);
+
+        Self(world, id)
+    }
+
+    #[inline(never)]
```

Comment:
Why did you choose to add `#[inline(never)]`? It seems counter-intuitive to me, since we're just trying to benchmark the contained `run_system()`.